### PR TITLE
feat(client): add mock service worker contents

### PR DESCRIPTION
**Pull request by joaoprcosta**
## Description

- Remove moxios inside contents
- add msw to fixtures

Refs #18 

### Dependencies

None.

## Checklist

- [X] The commit message follows our guidelines
- [X] Tests for the respective changes have been added
- [X] The code is commented, particularly in hard-to-understand areas
- [X] The labels and/or milestones were added

## Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

File: packages/client/src/contents/__fixtures__/commercepages.fixtures.js [hunk 1]
```diff
@@ -1,48 +1,27 @@
...
-        }),
-        {
-          response: params.response,
-          status: 200,
-        },
-      );
-    },
+    success: response =>
+      rest.get(path, async (req, res, ctx) =>
+        res(ctx.status(200), ctx.json({ ...response })),
```
Comment (helderburato) [inline packages/client/src/contents/__fixtures__/commercepages.fixtures.js:17]: I think you don't need to destructure `{ ...response }` you can just pass response as an argument to `ctx.join(...)`:
```js
ctx.join(response)
```

File: packages/client/src/contents/__fixtures__/contentTypes.fixtures.js [hunk 2]
```diff
@@ -1,43 +1,29 @@
...
-          response: params.response,
-          status: 200,
-        },
-      );
-    },
+    success: response =>
+      rest.get(
+        '/api/content/v1/spaces/:spaceCode/contentTypes',
+        async (req, res, ctx) =>
+          res(ctx.status(200), ctx.json({ ...response })),
```
Comment (helderburato) [inline packages/client/src/contents/__fixtures__/contentTypes.fixtures.js:17]: Same as above.

File: packages/client/src/contents/__tests__/getCommercePages.test.js [hunk 3]
```diff
@@ -1,18 +1,15 @@
...
 
 describe('getCommercePages()', () => {
   const expectedConfig = undefined;
 
   beforeEach(() => {
-    moxios.install(client);
     jest.clearAllMocks();
   });
 
-  afterEach(() => moxios.uninstall(client));
```
Comment (helderburato) [inline packages/client/src/contents/__tests__/getCommercePages.test.js:14]: I didn't tested, but I think you can call `resetHandlers()` here:
```js
afterEach(() => {
  mswServer.resetHandlers();
})
```

As mentioned in the [MSW docs](https://mswjs.io/docs/api/setup-server/reset-handlers#default-behavior):
_The .resetHandlers() function is useful as a clean up mechanism between multiple test suites that leverage runtime request handlers._

**Review by helderburato — APPROVED** [review]
Great work 🚀

File: tests/mswServer.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+import { setupServer } from 'msw/node';
+
+export const mswServer = setupServer();
```
Comment (nelsonleite) [inline tests/mswServer.js:3]: I think I'd replace this for a default export, feels more natural to me when importing, since this is a single setup 🤔 

```js
const mswServer = setupServer();

export default mswServer;
```

File: tests/setupAfterEnv.js [hunk 5]
```diff
@@ -2,3 +2,8 @@ import '@testing-library/jest-dom/extend-expect';
 import 'core-js/stable';
 import 'jest-axe/extend-expect';
 import 'regenerator-runtime/runtime';
+import { mswServer } from './mswServer';
+
+beforeAll(() => mswServer.listen());
```
Comment (nelsonleite) [inline tests/setupAfterEnv.js:7]: Won't this run everywhere, including places where this doesn't make sense, for example the react package?

Is it worth it, or is it useful anyway for every single test?

**Review by nelsonleite — CHANGES_REQUESTED** [review]

**Review by mindouro — APPROVED** [review]

**Review by ricardojmf — APPROVED** [review]

File: packages/client/src/contents/__fixtures__/commercepages.fixtures.js [hunk 6]
```diff
@@ -1,48 +1,27 @@
...
  * Response payloads.
  */
 export default {
   get: {
     /**
-     * Success moxios request.
+     * Success msw request.
      *
-     * @param {RequestParams} params - Params to moxios request.
+     * @param {Array} response - Commerce Pages payload.
```
Comment (boliveira) [inline packages/client/src/contents/__fixtures__/commercepages.fixtures.js:13]: Response seems to be an object and not an array. Can you check if that is the expected type?

File: packages/client/src/contents/__fixtures__/contentTypes.fixtures.js [hunk 7]
```diff
@@ -1,43 +1,29 @@
...
  * Response payloads.
  */
 export default {
   get: {
     /**
-     * Success moxios request.
+     * Success msw request.
      *
-     * @param {RequestParams} params - Params to moxios request.
+     * @param {Array} response - Content types payload.
```
Comment (boliveira) [inline packages/client/src/contents/__fixtures__/contentTypes.fixtures.js:11]: Same as https://github.com/Farfetch/blackout/pull/130/files#r786201319

File: packages/client/src/helpers/client/interceptors/authentication/__tests__/AxiosAuthenticationTokenManager.test.js [hunk 8]
```diff
@@ -679,10 +679,9 @@ describe('AxiosAuthenticationTokenManager', () => {
     it('should throw an error if a request is performed before token manager is loaded', () => {
       tokenManagerInstance = createAndSetTokenManagerInstance(client);
 
-      expect(tokenManagerInstance.isLoaded).toBeFalsy();
-
-      expect.assertions(1);
+      expect.assertions(2);
```
Comment (boliveira) [inline packages/client/src/helpers/client/interceptors/authentication/__tests__/AxiosAuthenticationTokenManager.test.js:682]: Question: Why was this necessary?

File: tests/setupAfterEnv.js (hunk 5 above)
Comment (nelsonleite) [reply tests/setupAfterEnv.js:7]: Wild suggestion - we could have this isolated and in every test needed, we just import and use `mswServerSetup()` 🪄

File: packages/client/src/helpers/client/interceptors/authentication/__tests__/AxiosAuthenticationTokenManager.test.js (hunk 8 above)
Comment (joaoprcosta) [reply packages/client/src/helpers/client/interceptors/authentication/__tests__/AxiosAuthenticationTokenManager.test.js:682]: This test failed after msw config. But, if you check, there are two expects in this test. I'm curious how did the test pass before this configuration...

File: packages/client/src/contents/__fixtures__/commercepages.fixtures.js (hunk 1 above)
Comment (joaoprcosta) [reply packages/client/src/contents/__fixtures__/commercepages.fixtures.js]: Done

File: tests/mswServer.js (hunk 4 above)
Comment (joaoprcosta) [reply tests/mswServer.js]: Done

File: tests/setupAfterEnv.js (hunk 5 above)
Comment (joaoprcosta) [reply tests/setupAfterEnv.js]: Could you check please?

File: packages/client/src/contents/__fixtures__/commercepages.fixtures.js (hunk 6 above)
Comment (joaoprcosta) [reply packages/client/src/contents/__fixtures__/commercepages.fixtures.js]: Done.

File: packages/client/src/contents/__fixtures__/contentTypes.fixtures.js (hunk 7 above)
Comment (joaoprcosta) [reply packages/client/src/contents/__fixtures__/contentTypes.fixtures.js]: Done.

File: packages/client/src/helpers/client/interceptors/authentication/__tests__/AxiosAuthenticationTokenManager.test.js (hunk 8 above)
Comment (joaoprcosta) [reply packages/client/src/helpers/client/interceptors/authentication/__tests__/AxiosAuthenticationTokenManager.test.js]: After the new config setup for msw, one assertion was only needed but this is strange because you have two expects in the same test.

Comment (boliveira) [reply packages/client/src/helpers/client/interceptors/authentication/__tests__/AxiosAuthenticationTokenManager.test.js]: Interesting question. I dig a little deeper in jest to understand why the test was failing now when it was passing before and the thing is, the added `afterEach` call to setupAfterEnv.js will change the timings internally in jest, namely when it will process the assertions. Without the `afterEach` call it will process the results right before the call to expect(getUser()).rejects... which will reset the internal assertions number to 0 while with the `afterEach` call it will only do that after all expects are done.

TLDR; 
The test code was in fact incorrect, it should be `expect.assertions(2)` as you have done to avoid these timings problems in jest depending on the kind of tests being done and the afterEach blocks defined. Closing it... 

Thanks!

**Review by nelsonleite — APPROVED** [review]
